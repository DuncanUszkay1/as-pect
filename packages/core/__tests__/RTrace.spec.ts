import { TestContext } from "../src/test/TestContext";
import { createRTraceModule } from "./setup/createRTraceModule";

let ctx: TestContext;

let start = new Promise<void>((resolve, reject) => {
  createRTraceModule({}, (err, result) => {
    if (err) {
      console.log(err);
      reject(err);
    } else {
      ctx = result!;
      resolve();
    }
  });
});

beforeEach(() => start);

describe("RTrace output", () => {
  test("Overall Statistics", () => {
    expect(ctx.freeCount).toMatchSnapshot("freeCount");
    expect(ctx.allocationCount).toMatchSnapshot("allocationCount");
    expect(ctx.errors).toMatchSnapshot("errors"); // matches empty array
  });

  for (const group of ctx.testGroups) {
    test(`Group: ${group.name}`, () => {
      expect(group.freeCount).toMatchSnapshot(`freeCount`);
      expect(group.allocationCount).toMatchSnapshot(`allocationCount`);
      expect(group.incrementCount).toMatchSnapshot(`incrementCount`);
      expect(group.decrementCount).toMatchSnapshot(`decrementCount`);
      expect(group.reallocationCount).toMatchSnapshot(`reallocationCount`);
      expect(group.rtraceStart).toMatchSnapshot(`rtraceStart`);
      expect(group.rtraceEnd).toMatchSnapshot(`rtraceEnd`);
      expect(group.rtraceDelta).toMatchSnapshot(`rtraceDelta`);
      expect(group.errors).toMatchSnapshot(`errors`); // matches empty array
    });

    describe(`Group: ${group.name}`, () => {
      for (const groupTest of group.tests) {
        test(`Test: ${groupTest.name}`, () => {
          expect(groupTest.freeCount).toMatchSnapshot(`freeCount`);
          expect(groupTest.allocationCount).toMatchSnapshot(`allocationCount`);
          expect(groupTest.incrementCount).toMatchSnapshot(`incrementCount`);
          expect(groupTest.decrementCount).toMatchSnapshot(`decrementCount`);
          expect(groupTest.reallocationCount).toMatchSnapshot(
            `reallocationCount`,
          );
          expect(groupTest.rtraceStart).toMatchSnapshot(`rtraceStart`);
          expect(groupTest.rtraceEnd).toMatchSnapshot(`rtraceEnd`);
          expect(groupTest.rtraceDelta).toMatchSnapshot(`rtraceDelta`);
          expect(groupTest.errors).toMatchSnapshot(`errors`); // matches empty array
        });
      }
    });
  }
});
