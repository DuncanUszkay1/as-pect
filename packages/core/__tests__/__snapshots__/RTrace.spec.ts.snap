// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RTrace output Group: RTrace api Test: RTrace.activeBlocks(): allocationCount 1`] = `2`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeBlocks(): decrementCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeBlocks(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeBlocks(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeBlocks(): incrementCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeBlocks(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeBlocks(): rtraceDelta 1`] = `2`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeBlocks(): rtraceEnd 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeBlocks(): rtraceStart 1`] = `10`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeGroupBlocks(): allocationCount 1`] = `2`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeGroupBlocks(): decrementCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeGroupBlocks(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeGroupBlocks(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeGroupBlocks(): incrementCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeGroupBlocks(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeGroupBlocks(): rtraceDelta 1`] = `2`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeGroupBlocks(): rtraceEnd 1`] = `10`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeGroupBlocks(): rtraceStart 1`] = `8`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeTestBlocks(): allocationCount 1`] = `2`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeTestBlocks(): decrementCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeTestBlocks(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeTestBlocks(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeTestBlocks(): incrementCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeTestBlocks(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeTestBlocks(): rtraceDelta 1`] = `2`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeTestBlocks(): rtraceEnd 1`] = `8`;

exports[`RTrace output Group: RTrace api Test: RTrace.activeTestBlocks(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.allocations(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.allocations(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.allocations(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.allocations(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.allocations(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.allocations(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.allocations(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.allocations(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.allocations(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.classIdOf(pointer: usize): allocationCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.classIdOf(pointer: usize): decrementCount 1`] = `2`;

exports[`RTrace output Group: RTrace api Test: RTrace.classIdOf(pointer: usize): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.classIdOf(pointer: usize): freeCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.classIdOf(pointer: usize): incrementCount 1`] = `2`;

exports[`RTrace output Group: RTrace api Test: RTrace.classIdOf(pointer: usize): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.classIdOf(pointer: usize): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.classIdOf(pointer: usize): rtraceEnd 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.classIdOf(pointer: usize): rtraceStart 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.count(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.count(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.count(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.count(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.count(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.count(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.count(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.count(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.count(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.decrements(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.decrements(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.decrements(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.decrements(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.decrements(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.decrements(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.decrements(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.decrements(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.decrements(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.frees(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.frees(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.frees(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.frees(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.frees(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.frees(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.frees(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.frees(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.frees(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupAllocations(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupAllocations(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupAllocations(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupAllocations(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupAllocations(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupAllocations(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupAllocations(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupAllocations(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupAllocations(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupDecrements(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupDecrements(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupDecrements(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupDecrements(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupDecrements(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupDecrements(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupDecrements(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupDecrements(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupDecrements(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupFrees(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupFrees(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupFrees(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupFrees(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupFrees(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupFrees(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupFrees(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupFrees(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupFrees(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupIncrements(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupIncrements(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupIncrements(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupIncrements(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupIncrements(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupIncrements(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupIncrements(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupIncrements(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupIncrements(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupReallocations(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupReallocations(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupReallocations(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupReallocations(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupReallocations(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupReallocations(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupReallocations(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupReallocations(): rtraceEnd 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.groupReallocations(): rtraceStart 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.increments(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.increments(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.increments(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.increments(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.increments(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.increments(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.increments(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.increments(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.increments(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.reallocations(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.reallocations(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.reallocations(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.reallocations(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.reallocations(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.reallocations(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.reallocations(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.reallocations(): rtraceEnd 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.reallocations(): rtraceStart 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.sizeOf(pointer: usize): allocationCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.sizeOf(pointer: usize): decrementCount 1`] = `3`;

exports[`RTrace output Group: RTrace api Test: RTrace.sizeOf(pointer: usize): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.sizeOf(pointer: usize): freeCount 1`] = `1`;

exports[`RTrace output Group: RTrace api Test: RTrace.sizeOf(pointer: usize): incrementCount 1`] = `3`;

exports[`RTrace output Group: RTrace api Test: RTrace.sizeOf(pointer: usize): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.sizeOf(pointer: usize): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.sizeOf(pointer: usize): rtraceEnd 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.sizeOf(pointer: usize): rtraceStart 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.testAllocations(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testAllocations(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testAllocations(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.testAllocations(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testAllocations(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testAllocations(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testAllocations(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testAllocations(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.testAllocations(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.testDecrements(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testDecrements(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testDecrements(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.testDecrements(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testDecrements(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testDecrements(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testDecrements(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testDecrements(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.testDecrements(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.testFrees(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testFrees(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testFrees(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.testFrees(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testFrees(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testFrees(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testFrees(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testFrees(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.testFrees(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.testIncrements(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testIncrements(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testIncrements(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.testIncrements(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testIncrements(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testIncrements(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testIncrements(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testIncrements(): rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.testIncrements(): rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace api Test: RTrace.testReallocations(): allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testReallocations(): decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testReallocations(): errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api Test: RTrace.testReallocations(): freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testReallocations(): incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testReallocations(): reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testReallocations(): rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace api Test: RTrace.testReallocations(): rtraceEnd 1`] = `12`;

exports[`RTrace output Group: RTrace api Test: RTrace.testReallocations(): rtraceStart 1`] = `12`;

exports[`RTrace output Group: RTrace api: allocationCount 1`] = `8`;

exports[`RTrace output Group: RTrace api: decrementCount 1`] = `8`;

exports[`RTrace output Group: RTrace api: errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace api: freeCount 1`] = `2`;

exports[`RTrace output Group: RTrace api: incrementCount 1`] = `8`;

exports[`RTrace output Group: RTrace api: reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace api: rtraceDelta 1`] = `6`;

exports[`RTrace output Group: RTrace api: rtraceEnd 1`] = `12`;

exports[`RTrace output Group: RTrace api: rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace tests Test: should allocate a bunch of objects: allocationCount 1`] = `3`;

exports[`RTrace output Group: RTrace tests Test: should allocate a bunch of objects: decrementCount 1`] = `6`;

exports[`RTrace output Group: RTrace tests Test: should allocate a bunch of objects: errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace tests Test: should allocate a bunch of objects: freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should allocate a bunch of objects: incrementCount 1`] = `9`;

exports[`RTrace output Group: RTrace tests Test: should allocate a bunch of objects: reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should allocate a bunch of objects: rtraceDelta 1`] = `3`;

exports[`RTrace output Group: RTrace tests Test: should allocate a bunch of objects: rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace tests Test: should allocate a bunch of objects: rtraceStart 1`] = `3`;

exports[`RTrace output Group: RTrace tests Test: should allocate an object: allocationCount 1`] = `1`;

exports[`RTrace output Group: RTrace tests Test: should allocate an object: decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should allocate an object: errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace tests Test: should allocate an object: freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should allocate an object: incrementCount 1`] = `1`;

exports[`RTrace output Group: RTrace tests Test: should allocate an object: reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should allocate an object: rtraceDelta 1`] = `1`;

exports[`RTrace output Group: RTrace tests Test: should allocate an object: rtraceEnd 1`] = `4`;

exports[`RTrace output Group: RTrace tests Test: should allocate an object: rtraceStart 1`] = `3`;

exports[`RTrace output Group: RTrace tests Test: should count references: allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should count references: decrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should count references: errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace tests Test: should count references: freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should count references: incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should count references: reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should count references: rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should count references: rtraceEnd 1`] = `6`;

exports[`RTrace output Group: RTrace tests Test: should count references: rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace tests Test: should deallocate all the objects: allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should deallocate all the objects: decrementCount 1`] = `4`;

exports[`RTrace output Group: RTrace tests Test: should deallocate all the objects: errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace tests Test: should deallocate all the objects: freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should deallocate all the objects: incrementCount 1`] = `4`;

exports[`RTrace output Group: RTrace tests Test: should deallocate all the objects: reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should deallocate all the objects: rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should deallocate all the objects: rtraceEnd 1`] = `7`;

exports[`RTrace output Group: RTrace tests Test: should deallocate all the objects: rtraceStart 1`] = `7`;

exports[`RTrace output Group: RTrace tests Test: should deallocate an object: allocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should deallocate an object: decrementCount 1`] = `1`;

exports[`RTrace output Group: RTrace tests Test: should deallocate an object: errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace tests Test: should deallocate an object: freeCount 1`] = `1`;

exports[`RTrace output Group: RTrace tests Test: should deallocate an object: incrementCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should deallocate an object: reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should deallocate an object: rtraceDelta 1`] = `-1`;

exports[`RTrace output Group: RTrace tests Test: should deallocate an object: rtraceEnd 1`] = `3`;

exports[`RTrace output Group: RTrace tests Test: should deallocate an object: rtraceStart 1`] = `4`;

exports[`RTrace output Group: RTrace tests Test: should start a label: allocationCount 1`] = `1`;

exports[`RTrace output Group: RTrace tests Test: should start a label: decrementCount 1`] = `2`;

exports[`RTrace output Group: RTrace tests Test: should start a label: errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace tests Test: should start a label: freeCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should start a label: incrementCount 1`] = `3`;

exports[`RTrace output Group: RTrace tests Test: should start a label: reallocationCount 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: should start a label: rtraceDelta 1`] = `1`;

exports[`RTrace output Group: RTrace tests Test: should start a label: rtraceEnd 1`] = `7`;

exports[`RTrace output Group: RTrace tests Test: should start a label: rtraceStart 1`] = `6`;

exports[`RTrace output Group: RTrace tests Test: trigger a reallocation: allocationCount 1`] = `6`;

exports[`RTrace output Group: RTrace tests Test: trigger a reallocation: decrementCount 1`] = `5`;

exports[`RTrace output Group: RTrace tests Test: trigger a reallocation: errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace tests Test: trigger a reallocation: freeCount 1`] = `6`;

exports[`RTrace output Group: RTrace tests Test: trigger a reallocation: incrementCount 1`] = `5`;

exports[`RTrace output Group: RTrace tests Test: trigger a reallocation: reallocationCount 1`] = `1`;

exports[`RTrace output Group: RTrace tests Test: trigger a reallocation: rtraceDelta 1`] = `0`;

exports[`RTrace output Group: RTrace tests Test: trigger a reallocation: rtraceEnd 1`] = `7`;

exports[`RTrace output Group: RTrace tests Test: trigger a reallocation: rtraceStart 1`] = `7`;

exports[`RTrace output Group: RTrace tests: allocationCount 1`] = `11`;

exports[`RTrace output Group: RTrace tests: decrementCount 1`] = `18`;

exports[`RTrace output Group: RTrace tests: errors 1`] = `Array []`;

exports[`RTrace output Group: RTrace tests: freeCount 1`] = `7`;

exports[`RTrace output Group: RTrace tests: incrementCount 1`] = `22`;

exports[`RTrace output Group: RTrace tests: reallocationCount 1`] = `1`;

exports[`RTrace output Group: RTrace tests: rtraceDelta 1`] = `4`;

exports[`RTrace output Group: RTrace tests: rtraceEnd 1`] = `7`;

exports[`RTrace output Group: RTrace tests: rtraceStart 1`] = `3`;

exports[`RTrace output Group: a group reference counting error Test: empty: allocationCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error Test: empty: decrementCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error Test: empty: errors 1`] = `Array []`;

exports[`RTrace output Group: a group reference counting error Test: empty: freeCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error Test: empty: incrementCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error Test: empty: reallocationCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error Test: empty: rtraceDelta 1`] = `0`;

exports[`RTrace output Group: a group reference counting error Test: empty: rtraceEnd 1`] = `0`;

exports[`RTrace output Group: a group reference counting error Test: empty: rtraceStart 1`] = `0`;

exports[`RTrace output Group: a group reference counting error: allocationCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error: decrementCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error: errors 1`] = `Array []`;

exports[`RTrace output Group: a group reference counting error: freeCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error: incrementCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error: reallocationCount 1`] = `0`;

exports[`RTrace output Group: a group reference counting error: rtraceDelta 1`] = `0`;

exports[`RTrace output Group: a group reference counting error: rtraceEnd 1`] = `0`;

exports[`RTrace output Group: a group reference counting error: rtraceStart 1`] = `7`;

exports[`RTrace output Overall Statistics: allocationCount 1`] = `22`;

exports[`RTrace output Overall Statistics: errors 1`] = `Array []`;

exports[`RTrace output Overall Statistics: freeCount 1`] = `10`;
